design-documents/graph-ql/coverage/catalog/remove-product-attributes.md

```diff
@@ -0,0 +1,45 @@
+# Remove product dynamic attributes from ProductInterface
+
+## Current state
+
+Currently, Magento exposes EAV attributes as part of the product schema as top-level fields of ProductInterface. As consequences:
+* The merchant can change the GraphQL from the admin, which makes it hard to cache GraphQL schema.
+* Regular product interface overloaded with fields and almost unreadable.
+* It is quite often practice to create product programmatically, so the number of attributes could be really excessive with tens thousands of attributes.
+* Two Magento instances do not have the same GraphQL schema. As a result, it is near impossible to build a simple client/SDK, which could be easily transferred from one instance to another.
+* It is impossible to aggregate the several Magento instances behind the common [BFF](https://docs.microsoft.com/en-us/azure/architecture/patterns/backends-for-frontends).
```

Comment:
Extensions can add new attributes on the top level too. Why do we need to hide several Magento instances behind common BFF?

Reply:
for instance, if we need to add a common gateway with query cache and want to make this gateway saas solution.